src/Engine/GameEvents/CaptureEvent.java

```diff
@@ -54,14 +54,20 @@ public boolean willCapture()
    * until we change this by introducing new, game-breaking mechanics).
    */
   @Override
-  public void performEvent(GameMap gameMap)
+  public void performEvent(MapMaster gameMap)
   {
     // Only attempt to do the action if it is valid to do so.
     if( location.isCaptureable() &&
-        (location.getResident() == unit) &&
+        (location.getResident(gameMap) == unit) &&
         (unit.CO.isEnemy(location.getOwner())) )
     {
       unit.capture(gameMap.getLocation(unit.x, unit.y));
     }
   }
+  
+  @Override // there's no known way for this to fail after the GameAction is constructed
+  public boolean shouldPreempt(MapMaster gameMap )
+  {
+    return false;
```

Comment:
It can *be* preempted, but it cannot itself preempt.
The move action can fail (which will throw a shouldPreempt), which will cause the CaptureEvent to not happen... but the CaptureEvent only governs what happens once you're already on top of the space to be captured.

Reply:
Oh, right. `CaptureEvent` cannot be preempted, but `CaptureAction` can be. I was moving the checks for preemption to the action side, and got a wire crossed in my head.